misc/qlog-adapter.py

```diff
@@ -328,22 +328,32 @@ def render_ack_frame(ranges):
 def usage():
     print(r"""
 Usage:
-    python qlog-adapter.py inTrace.jsonl
+    python3 qlog-adapter.py [inTrace.jsonl]
+
+    If the argument is omitted, inTrace will be read from stdin.
 """.strip())
 
 def load_quicly_events(infile):
     events = []
-    with open(infile, "r") as fh:
-        for line in fh:
-            events.append(json.loads(line))
+    for line in infile:
+        events.append(json.loads(line))
     return events
 
 def main():
-    if len(sys.argv) != 2:
+    if len(sys.argv) > 2:
         usage()
         sys.exit(1)
 
-    (_, infile) = sys.argv
+    if len(sys.argv) == 1:
+        infile = sys.stdin
+    else:
+        (_, fn) = sys.argv
+        try:
+            infile = open(fn, "r")
```

Comment:
Just realized that I'm not closing `infile` in this case (whereas in the original code it automatically does with `with`. However I'd say I don't care because it is opened for read (no fear of data loss due to buffer not being flushed) and one time thing in a short-lived command-line utility (no fear of leaked resource being accumulated).